tensorboard/data/server/data_compat.rs

```diff
@@ -671,6 +696,48 @@ mod tests {
             );
         }
 
+        #[test]
+        fn test_graph_subplugins() {
+            for &plugin_name in &[
+                GRAPH_RUN_METADATA_PLUGIN_NAME,
+                GRAPH_RUN_METADATA_WITH_GRAPH_PLUGIN_NAME,
+                GRAPH_KERAS_MODEL_PLUGIN_NAME,
+            ] {
+                let md = pb::SummaryMetadata {
+                    plugin_data: Some(PluginData {
+                        plugin_name: plugin_name.to_string(),
+                        content: b"1".to_vec(),
+                        ..Default::default()
+                    }),
+                    ..Default::default()
+                };
+                let v = SummaryValue(Box::new(Value::Tensor(pb::TensorProto {
+                    dtype: pb::DataType::DtString.into(),
+                    tensor_shape: Some(tensor_shape(&[])),
+                    string_val: vec![b"some-graph-proto".to_vec()],
+                    ..Default::default()
+                })));
+
+                // Test both metadata and enrichment here, for convenience.
+                let initial_metadata = v.initial_metadata(Some(md));
+                assert_eq!(
+                    *initial_metadata,
+                    pb::SummaryMetadata {
+                        plugin_data: Some(PluginData {
+                            plugin_name: plugin_name.to_string(),
+                            content: b"1".to_vec(),
+                            ..Default::default()
+                        }),
+                        data_class: pb::DataClass::BlobSequence.into(),
+                        ..Default::default()
+                    },
+                );
```

Comment:
No AI required: this test feels too mechanical :\

Reply:
Like, too much of “assert that `f(x)` equals the result of manually
computing `f(x)`”? I agree; it does. I think that I feel okay with it
since it is actually meaningful coverage—dropping a single line in the
`match` would cause parts of the graph plugin to silently not render.

Or maybe you meant that the test implementation spends a lot of lines on
proto manipulation? Agreed there, too. We could refactor to reuse proto
sub-structures, but I generally like optimizing for readability in tests
like this. In #4552, for example, I made the tests more explicit because
if they had been written like that in the first place, the bug would
likely have been caught in code review: `new Date(1610613017)` is an
obvious bug—too few digits—whereas `new Date(fakePoints1[0].x)` is not
obviously correct and just repeats the implementation code, which was
also wrong.

Noted that you said “no AI required”, so feel free to reply or not, as
you like. Glad that this passes your minimum bar, but also always
interested in doing better than that.